File: sdk/runanywhere-flutter/lib/core/models/framework/framework_modality.dart
/// Represents the different modalities (capabilities) a framework can support.
///
/// This is the Flutter equivalent of Swift's `FrameworkModality`.
enum FrameworkModality {
  /// Speech-to-text (STT) - voice input to text output
  voiceToText('voice_to_text'),

  /// Text-to-speech (TTS) - text input to voice output
  textToVoice('text_to_voice'),

  /// Text-to-text (LLM) - text generation
  textToText('text_to_text'),

  /// Vision-to-text (VLM) - image input to text output
  visionToText('vision_to_text'),

  /// Image-to-text - image understanding
  imageToText('image_to_text'),

  /// Text-to-image - image generation
  textToImage('text_to_image'),

  /// Multimodal - multiple input/output types
  multimodal('multimodal'),

  /// Voice activity detection (VAD)
  voiceActivityDetection('voice_activity_detection'),

  /// Speaker diarization - identify who is speaking
  speakerDiarization('speaker_diarization'),

  /// Wake word detection
  wakeWord('wake_word'),

  /// Text embeddings
  textEmbedding('text_embedding');

  const FrameworkModality(this.rawValue);

  /// The string representation of this modality.
  final String rawValue;

  /// Create from raw value string.
  static FrameworkModality? fromRawValue(String value) {
    for (final modality in values) {
      if (modality.rawValue == value) {
        return modality;
      }
    }
    return null;
  }
}

File: sdk/runanywhere-react-native/src/Core/Models/Framework/FrameworkModality.ts
/**
 * FrameworkModality.ts
 *
 * Defines the input/output modalities a framework supports
 *
 * Reference: sdk/runanywhere-swift/Sources/RunAnywhere/Core/Models/Framework/FrameworkModality.swift
 */

/**
 * Defines the input/output modalities a framework supports
 */
export enum FrameworkModality {
  TextToText = 'text-to-text', // Traditional LLM text generation
  VoiceToText = 'voice-to-text', // Speech recognition/transcription
  TextToVoice = 'text-to-voice', // Text-to-speech synthesis
  ImageToText = 'image-to-text', // Image captioning/OCR
  TextToImage = 'text-to-image', // Image generation
  Multimodal = 'multimodal', // Supports multiple modalities
}

/**
 * Human-readable display name for a modality
 */
export function getFrameworkModalityDisplayName(modality: FrameworkModality): string {
  switch (modality) {
    case FrameworkModality.TextToText:
      return 'Text Generation';
    case FrameworkModality.VoiceToText:
      return 'Speech Recognition';
    case FrameworkModality.TextToVoice:
      return 'Text-to-Speech';
    case FrameworkModality.ImageToText:
      return 'Image Understanding';
    case FrameworkModality.TextToImage:
      return 'Image Generation';
    case FrameworkModality.Multimodal:
      return 'Multimodal';
  }
}

/**
 * Icon name for UI display
 */
export function getFrameworkModalityIconName(modality: FrameworkModality): string {
  switch (modality) {
    case FrameworkModality.TextToText:
      return 'text.bubble';
    case FrameworkModality.VoiceToText:
      return 'mic';
    case FrameworkModality.TextToVoice:
      return 'speaker.wave.2';
    case FrameworkModality.ImageToText:
      return 'photo.badge.arrow.down';
    case FrameworkModality.TextToImage:
      return 'photo.badge.plus';
    case FrameworkModality.Multimodal:
      return 'sparkles';
  }
}

